### ISSUE TITLE
Better error reporting when CSS is bad/invalid

### ISSUE THREAD

**Issue by earksiinni** [styled-components/styled-components#1933]
tl;dr: Reporting CSS errors, even something simple like printing "there's a problem with your CSS" in the console without any line numbers or other references, would be a massive aid during development.

I love styled components, I really do. But a huge pain for my team is that whenever we write bad CSS, styled components seems to fail silently. A recent example from a project we were working on where we were converting CSS/SASS files:

```
const globalStyles = {
  mediaQueries: {
    wide: wide: `screen and (min-width: 1200px)`
  }
}

styled.div`
  @media #{${globalStyles.mediaQueries.wide}}
`
```

The problem here is the use of the SASS interpolation `#{...}` in `styled.div`. When we started refactoring this particular component, we copied and pasted the rules from the SASS file, and this particular media query made use of a SASS interpolation that looked something like `@media #{$wideMediaQuery}`. We replaced the SASS variable with our own interpolation, but we forgot to remove the surrounding SASS interpolation. Styled components silently broke and failed to generate the style tag. Stylelint didn't catch this particular error, either, and even if it did, one shouldn't exclusively rely on a linter for error reporting.

I could provide countless other examples. Probably the most common is forgetting to put a semicolon at the end of a rule, which will result in an empty style and no error.

On a related note, does anyone have tips/tricks/workarounds to get better error reporting?

**Comment by probablyup** [comment]
We have a stylint setup for this since it would be impractical to do CSS
syntax and property-value validity sort of things at runtime.
https://www.styled-components.com/docs/tooling#stylelint

**Comment by earksiinni** [comment]
@probablyup, as I mentioned, stylelint doesn't catch everything including the example that I just mentioned. And I don't see in principle why validation would be hard. Plenty of CSS validators exist. Running the compiled CSS against one of them and simply saying "you have invalid CSS somewhere" would be a vast improvement over the current status quo, which is to fail very silently and inconsistently.

**Comment by earksiinni** [comment]
Not to be like "why can't you be like this much better funded and staffed project," but in my mind I was thinking something like dev mode in React.

**Comment by probablyup** [comment]
It's the best we can do for the moment, sorry. You're welcome to explore
this sort of thing if you'd like, possibly as a stylis plugin.

**Comment by mxstbr** [comment]
Yeah I'd like this as a dev-mode enhancement!
